tools/fuzzer.cpp

```diff
@@ -0,0 +1,274 @@
+  static auto &availableChars = "0123456789"
+                                "abcdefghijklmnopqrstuvwxyz"
+                                "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  size_t length = randomGenerator.mt64() % 1000;
+  std::string s;
+  s.reserve(length);
+  while (length--) {
+    s += availableChars[randomGenerator.mt64() % (sizeof(availableChars) - 1)];
+  }
+  builder.add(Value(s));
+}
+
+static void generateVelocypack(Builder& builder, size_t depth, RandomGenerator& randomGenerator, Format const& format) {
+  while (true) {
+    RandomBuilderAdditions randomBuilderAdds = static_cast<RandomBuilderAdditions>(randomGenerator.mt64() %
+                                                                                   (format == Format::JSON
+                                                                                    ? ADD_DOUBLE
+                                                                                    : ADD_MAX_VPACK_VALUE));
+    if (depth > 10 && randomBuilderAdds <= ADD_OBJECT) {
+      continue;
+    }
+    switch (randomBuilderAdds) {
+      case ADD_ARRAY: {
+        builder.openArray(randomGenerator.mt64() % 2 ? true : false);
+        size_t numMembers = randomGenerator.mt64() % 10;
+        for (size_t i = 0; i < numMembers; ++i) {
+          generateVelocypack(builder, depth + 1, randomGenerator, format);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_OBJECT: {
+        builder.openObject(randomGenerator.mt64() % 2 ? true : false);
+        size_t numMembers = randomGenerator.mt64() % 10;
+        for (size_t i = 0; i < numMembers; ++i) {
+          std::string key = "test" + std::to_string(i);
+          builder.add(Value(key));
+          generateVelocypack(builder, depth + 1, randomGenerator, format);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_BOOLEAN:
+        builder.add(Value(randomGenerator.mt64() % 2 ? true : false));
+        break;
+      case ADD_STRING:
+        addString(builder, randomGenerator);
+        break;
+      case ADD_NULL:
+        builder.add(Value(ValueType::Null));
+        break;
+      case ADD_UINT64:
+        builder.add(Value(randomGenerator.mt64()));
+        break;
+      case ADD_INT64: {
+        uint64_t uintValue = randomGenerator.mt64();
+        int64_t intValue;
+        memcpy(&intValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(intValue));
+        break;
+      }
+      case ADD_DOUBLE: { //exception was thrown saying JSON doesn't support this type
+        uint64_t uintValue = randomGenerator.mt64();
+        double doubleValue;
+        memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(doubleValue));
+        break;
+      }
+      case ADD_UTC_DATE:
+        builder.add(Value(randomGenerator.mt64(), ValueType::UTCDate));
+        break;
+      case ADD_BINARY:
+        builder.add(Value(std::bitset<64>(randomGenerator.mt64()).to_string()));
```

Comment:
Can we support variable-length binary values?

Reply:
By the way, this will not generate VelocyPack binary values, it will generate VelocyPack string values.